server/src/main/java/org/opensearch/index/mapper/DateFieldMapper.java

```diff
@@ -559,6 +568,9 @@ public Relation isFieldWithinQuery(
             DateMathParser dateParser,
             QueryRewriteContext context
         ) throws IOException {
+            if (isSearchable() == false && hasDocValues()) {
```

Comment:
Could you please explain this change?

Reply:
So if we know that we have a `doc_values` only query we can simply return intersects because we have no way of checking the bkd for the actual range, we have to iterate though all the doc ids for find the actual range

Reply:
May be worth a comment here?